### ISSUE TITLE
Prevent multiple compensation handlers per activity

### ISSUE THREAD

**Issue by nicpuppa** [camunda/camunda#19634]
**Describe the bug**

Zeebe is allowing customer to deploy an activity with multiple compensation handler (and multiple compensation boundary event) attached. The specification is not clear about this, but it uses singular form whenever a compensation handler is mentioned. But it doesn't make sense to have multiple compensation handler for a single activity.

C7 also reject the deployment of a process with multiple compensation boundary events

```
ENGINE-09005 Could not parse BPMN process. Errors: 
* multiple boundary events with compensateEventDefinition not supported on same activity | resource test.bpmn | line 22 | column 80 [ deploy-error ]
```

**To Reproduce**

Deploy a process with multiple compensation handler attached to a single activity.

![image_720](https://github.com/camunda/camunda/assets/72783243/c5d8683c-d27c-49f6-a987-181f1d05d685)

**Expected behavior**

Zeebe should reject the deployment with an error

related issue: https://github.com/camunda/camunda-modeler/issues/4395

**Comment by korthout** [comment]
ZPA triage:
- current impact: zeebe only picks up at the first handler, the second is ignored. Is unexpected from user's perspective
- no useful workaround available
- it's not useful to compensate twice
- existing deployed models will continue to suffer from this issue, it's only a protection against new models
- to avoid this new rule causing confusion when users make other changes in existing models that have this, we need to clearly communicate why we reject the model
- we could raise an incident for existing process instances with this problem, but we argue that it's better to not disrupt existing process instances now (breaking user-space in a patch). Unless we see support cases about this behavior
- likelihood: unknown, and users shouldn't typically have a good reason to do this

**Comment by eppdot** [comment]
Update (re-triaging with @remcowesterhoud)

@saig0 what are your thoughts on this? We are leaning towards aligning with the C7 behavior.

**Comment by saig0** [comment]
@eppdot I agree with the analysis and the declaration of the behavior as a *bug*. We should reject a process deployment with multiple compensation handlers for one activity. 

> we could raise an incident for existing process instances with this problem, but we argue that it's better to not disrupt existing process instances now (breaking user-space in a patch)

This is a good point. 👍 We must avoid breaking user-space, for example, by applying the validation only for new process models, but ignoring already deployed processes.

**Comment by marcosgvieira** [comment]
### 🧭 Triage Assessment

| Field | Value |
|------|-------|
| **Work type** | BUG |
| **Severity** | ⚠️ Mid |
| **Likelihood** | ⚠️ Mid |
| **Confidence** | 0.9 |

---

### 📝 Rationale

The issue describes Zeebe allowing deployment of an activity with multiple compensation handlers, which is unexpected and inconsistent with C7 behavior. The expected behavior is to reject such deployments. The impact is noticeable (unexpected user experience, no useful workaround, and existing models continue to suffer), but does not cause data loss or critical system failure, so severity is MID. The likelihood is MID, as the scope is unknown and users should not typically do this, but it is not a rare edge case. The issue is clearly within Core Features team boundaries.

---

🤖 _This issue was automatically triaged by an AI-based system as part of the Core Features triage workflow._

✋ **Want a human review?**
If you believe this assessment is incorrect or needs adjustment, feel free to update the details or request a manual triage by adding the **`triage:manual`** label.
